File: prog/1stPartyLibs/quirrel/quirrel/bench/js/primes.js
// primes test

function isprime(n) {
	var m = n - 1;
	for(var i=2; i!=m; ++i ) {
		if ( n % i ==0 ) {
			return false;
		}
	}
	return true;
}

function primes(n) {
	var count = 0;
	for ( var i=2; i!=n; ++i ) {
		if ( isprime(i) ) {
			count ++;
		}
	}
	return count;
}

// infrastructure

function timeStamp() {
	return Date.now();
}

function profile(tname,cnt,testFn) {
	var t = 100500;
	var count = cnt
	while ( count>0 ) {
		var t0 = timeStamp();
		testFn();
		var t1 = timeStamp();
		var dt = t1 - t0;
		t = Math.min(dt, t);
		count --;
	}
	t /= 1000.0;
	var msg = '"'+tname+'", '+t+', '+cnt;
	print(msg)
}

function performance_tests() {
	profile("primes loop",20,function(){
		primes(14000);
	});
	timeStamp();
}

performance_tests();

File: prog/1stPartyLibs/quirrel/quirrel/bench/lua/primes.lua
function isprime(n)
    for i = 2, (n - 1) do
        if (n % i == 0) then
            return false
        end
    end
    return true
end


local function primes(n)
    local count = 0

    for i = 2, n do
        if (isprime(i)) then
            count = count + 1
        end
    end
    return count
end

loadfile("profile.lua")()

io.write(string.format("\"primes loop\", %.8f, 20\n", profile_it(20, function () primes(14000) end)))

